### PR TITLE
Use dumb dbm driver for all platforms

### PR DESCRIPTION
**Pull request by nimnull**
## RATIONALE

https://datarobot.atlassian.net/browse/API-4967

`shelve` uses `dbm` implemented by a number of platform-dependent C extensions. Issue reproduced several times on MacOS with `dbm.ndbm` extension. However linux with dummy (pure-python fallback) that works for all cases.

## CHANGES

Manually specify `dbm` implementation type to make it platform-issues free.

## TESTING

Ticket contains sample continuously failing on MacOS at 6599 batches (1 row per batch) even if sample is shuffled. That change allows to score that full sample with no changes to the options.

### CODE REVIEW

File: datarobot_batch_scoring/writer.py [hunk 1]
```diff
@@ -10,6 +10,11 @@
 from functools import reduce
 from time import time
 
+try:
```
Comment (Axik) [inline datarobot_batch_scoring/writer.py]:
```python
if six.PY3:
    from dbm import dumb as dumb_dbm
else:
    import dumbdbm as dumb_dbm
```

**Comment by nimnull** [comment]
@Axik how about aprove?

**Comment by Axik** [comment]
I missed couple of things.
1. Bump version
2. Update CHANGES.rst

**Comment by nimnull** [comment]
@Axik done. Ready to merge & release

**Pull request merged**